_blog/scaling-dot-net-core-microservices/index.md

```diff
@@ -0,0 +1,403 @@
+
+So far, we've found out that running our microservice application has been a bit harder to do than we'd like.
+If we compare our services to spinning plates, we have a lot of spinning plates that are dependent on each other and we're not great at noticing when a plate has fallen.
+We're going to add a `Dockerfile` to each of our services to containerise them.
+
+### Containers should do just one thing, and one thing well
+
+Containers can be whatever you want them to be, a 30mb console application to a 4gb monolith.
+But they should really just do one job and do one job well.
+When you scale your application, do you want to scale your whole application or just some services within it?
+If a service fails or has an error, do you want to know that it was the one thing that did it, or possibly one or many of several?
+
+### Do I need to do anything differently if the code is in a container
+
+If you can build and run your code locally, you'll be able to run it within a container.
+If you're running an application and you have to manually kill threads locally, will this work within a container?
+Yes, it will!  But you're going to find problems down the line.
+
+The containers run microservices, if we implement bad code and practices, these will just grow when scaling the application.
+As with most concerns, by addressing the common issues early vast improvements can be attained later.
+
+> TOP TIP: When you build something, learn how to kill it and make it graceful
+
+Regardless of what part of your system this relates to, you should be able to create and delete parts of your system.
+For example, if you're creating a cluster, you should be able to have a command that can delete your cluster.
+If you create a console application, you should manage the `Ctrl + C` or `SIGTERM` event.
+Normally, containers will give containers 30 seconds notice before terminating an application.
+If you have a console application that is processing a lead, how do you manage this when the orchestrator fires the `Ctrl + C` event?
+Do you delete the process or wait for it to complete?  These are the decisions you should be making prior to moving to a container.
+Also, this will empower your team later on, as you'll know how to manage when you have an actual fire.
+
+### Docker Build
+
+We use Docker build to create Docker images.  Docker images can then be used to spin up containers.
+
+> TOP TIP: Environment Variables are your Best Friend Forever (BFF)
+
+When running containers, you have the option to pass variables into the container via a number of ways:
+
+* Volumes
+* Environment Variable
+
+When should you them?  Use volumes for static assets that might need to change.  For example, if you have a website then load your images via a volume, or if you have a database then keep the data within a volume so that you don't lose it when the container is deleted.
+Use Environment Variables for connections strings and any secrets that you're managing with the application.
```

Comment:
Not sure how this is related to what we're doing. Yes, I know how to set environment variables, but... why? I think you should remove this part since it's not relevant. We want to run the app(s), so the next section addresses that.